### DISCUSSION TITLE
Cutter Compensation

### DISCUSSION THREAD

**Discussion by mlewus** [terjeio/grblHAL#311] [Ideas]
Any plans to implement cutter compensation G40, G41, G42 for grblHAL on RT1062? This is not implemented in grbl but it is very often used for single point threading on milling machines. I looked at the existing code and I believe this is outside my capabilities but I think it would be a welcome addition. Anybody agree? I did a search on these g codes and I did not see any old discussions so either I missed it or no one has brought this up before.

**Comment by terjeio** [comment]
It is kind of planned as I have looked at how it could be implemented. IIRC a lookahead queue for motion commands needs to be implemented so that enough moves will be available for applying the math. I will be a lot of work, most of it testing?

It has been [discussed for grbl](https://github.com/grbl/grbl/issues/385) before, and there are links there to code that may be useful.

**Reply by mlewus** [reply]
I am within a week or two of finishing a project to convert a 4 axis Sherline desktop mill to CNC. It uses a i.MX RT1062 (Teensy 4.1) breakout board from @phil-barrett. I can assist with functional testing on RT1062 once the machine is up and running.

I can also assist with unit testing anytime if you decide to implement these features.
